lib/money/money.rb

```diff
@@ -340,14 +340,29 @@ def allocate_max_amounts(maximums)
   # @example
   #   Money.new(100, "USD").split(3) #=> [Money.new(34), Money.new(33), Money.new(33)]
   def split(num)
+    num_high, num_low, high, low = calculate_splits(num)
+    Array.new(num_high, high) + Array.new(num_low, low)
+  end
+
+  # Calculate the splits evenly without losing pennies.
+  # Returns the number of high and low splits and the value of the high and low splits.
+  # Where high represents the Money value with the extra penny
+  # and low a Money without the extra penny.
+  #
+  # @param [2] number of parties.
+  #
+  # @return [Array<Integer, Integer, Money, Money>]
+  #
+  # @example
+  #   Money.new(100, "USD").calculate_splits(3) #=> [1, 2, Money.new(34), Money.new(33)]
+  def calculate_splits(num)
     raise ArgumentError, "need at least one party" if num < 1
     subunits = self.subunits
     low = Money.from_subunits(subunits / num, currency)
     high = Money.from_subunits(low.subunits + 1, currency)
 
-    remainder = subunits % num
-
-    return Array.new(remainder, high) + Array.new(num - remainder, low)
+    num_high = subunits % num
+    [num_high, (num - num_high), high, low]
```

Comment:
what was the argument against 
```ruby
{ high => num_high, low => num_low }
```
the split method would become
```ruby
  def split(num)
    calculate_splits(num).sum { |value, count| Array.new(value, count) }
  end
```

Reply:
As long as you don't care to know which is which it would work.